src/Microsoft.AspNetCore.Routing/Matching/VectorizedRefEmitTrieJumpTable.cs

```diff
@@ -0,0 +1,548 @@
+            public LocalBuilder UInt16Value { get; set; }
+
+            /// <summary>
+            /// Holds current character when processing 4 characters at a time.
+            /// </summary>
+            public LocalBuilder UInt64Value { get; set; }
+
+            /// <summary>
+            /// Used to covert casing. See comments where it's used.
+            /// </summary>
+            public LocalBuilder UInt64LowerIndicator { get; set; }
+
+            /// <summary>
+            /// Used to covert casing. See comments where it's used.
+            /// </summary>
+            public LocalBuilder UInt64UpperIndicator { get; set; }
+
+            /// <summary>
+            /// Holds a 'ref byte' reference to the current character (in bytes).
+            /// </summary>
+            public LocalBuilder P { get; set; }
+
+            /// <summary>
+            /// Holds the relevant portion of the path as a Span[byte].
+            /// </summary>
+            public LocalBuilder Span { get; set; }
+        }
+
+        private class Labels
+        {
+            /// <summary>
+            /// Label to goto that will return the default destination (not a match).
+            /// </summary>
+            public Label ReturnDefault { get; set; }
+
+            /// <summary>
+            /// Label to goto that will return a sentinel value for non-ascii text.
+            /// </summary>
+            public Label ReturnNotAscii { get; set; }
+        }
+
+        private class Methods
+        {
+            public Methods()
+            {
+                // Can't use GetMethod because the parameter is a generic method parameters.
+                Add = typeof(Unsafe)
+                    .GetMethods(BindingFlags.Public | BindingFlags.Static)
+                    .Where(m => m.Name == nameof(Unsafe.Add))
+                    .Where(m => m.GetGenericArguments().Length == 1)
+                    .Where(m => m.GetParameters().Length == 2)
+                    .FirstOrDefault()
+                    ?.MakeGenericMethod(typeof(byte));
+                if (Add == null)
+                {
+                    throw new InvalidOperationException("Failed to find Unsafe.Add{T}(ref T, int)");
+                }
+
+                // Can't use GetMethod because the parameter is a generic method parameters.
+                As = typeof(Unsafe)
+                    .GetMethods(BindingFlags.Public | BindingFlags.Static)
+                    .Where(m => m.Name == nameof(Unsafe.As))
+                    .Where(m => m.GetGenericArguments().Length == 2)
+                    .Where(m => m.GetParameters().Length == 1)
+                    .FirstOrDefault()
+                    ?.MakeGenericMethod(typeof(char), typeof(byte));
+                if (Add == null)
+                {
+                    throw new InvalidOperationException("Failed to find Unsafe.As{TFrom, TTo}(ref TFrom)");
+                }
+
+                AsSpan = typeof(MemoryExtensions).GetMethod(
```

Comment:
If you want to make that GetMethod code more readable, you can cheat with Expression Trees:
```csharp
Expression<Action<string>> functor = s => s.AsSpan(0,0);
var asSpanMethodInfo = (functor.Body as MethodCallExpression).Method;
```

Reply:
I briefly thought about it, but this is already done 😆